### PR TITLE
arch/risc-v/mpfs: Remove duplicated riscv_restorefpu

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary

Follow #5868, forget to remove it from mpfs

## Impact
MPFS
## Testing
CI

### CODE REVIEW

**Comment by no1wudi** [comment]
@pussuw Could you take a look ?

**Review by pussuw — APPROVED** [review]
LGTM

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/risc-v/src/mpfs/mpfs_irq_dispatch.c [hunk 1]
```diff
@@ -149,9 +142,6 @@ void *riscv_dispatch_irq(uintptr_t vector, uintptr_t *regs)
       group_addrenv(NULL);
 #endif
     }
-#endif
-
-#endif
```
Comment (pussuw) [inline arch/risc-v/src/mpfs/mpfs_irq_dispatch.c:154]: I think one too many #ifdefs got removed. The build fails now (CI)

Comment (pussuw) [reply arch/risc-v/src/mpfs/mpfs_irq_dispatch.c:154]: LGTM

**Review by pussuw — APPROVED** [review]

**Review by pussuw — APPROVED** [review]

**Review by pussuw — APPROVED** [review]

**Pull request merged**